### PR TITLE
do not comment on PRs, instead store output as logs

### LINKED ISSUE #416: #406 followup: p-s should not comment on PRs when SRPM creation fails
We should store all those exceptions in DB as logs and provide them in the logs view.

TODO:
- change all the code which comments in PRs to append a log message inside DB

Follow-up to #406 

Related: #243 #168

### PR DESCRIPTION
**Pull request by TomasTomecek**
Related #168
Fixes #416

TODO:
* [x] kinda finish the code
* [x] unit tests are passing
* [x] fix integration tests
* [x] tests: create expectations for the new behaviour
* [ ] get this reviewed by @dhodovsk
* [x] get this reviewed by @lachmanfrantisek

I completely rewrote how we handle errors and failures during the srpm creation process - please let me know how I should polish it.

### CODE REVIEW

File: packit_service/worker/build/copr_build.py [hunk 1]
```diff
@@ -198,124 +177,50 @@ def _run_copr_build_and_save_output(self):
...
+            # all the other metadata, so... FIXME
+            c.srpm_failed = True
+        finally:
+            packit_logger.removeHandler(handler)
+            stream.seek(0)
+            c.srpm_logs = stream.read()
+        if c.srpm_failed:
+            # when do we NOT want to send stuff to sentry?
+            sentry_integration.send_to_sentry(ex)
+            if isinstance(ex, SandcastleTimeoutReached):
```
[the remark this answers is not in the document]
Comment (TomasTomecek) [reply packit_service/worker/build/copr_build.py]: please suggest a better way how to organize the code

[the remark this answers is not in the document]
Comment (lachmanfrantisek) [reply packit_service/worker/build/copr_build.py]: Use `try-except`?

```python
        try:
            c.copr_build_id, c.copr_web_url = self.api.run_copr_build(
                project=self.job_project,
                chroots=self.build_chroots,
                owner=self.job_owner,
            )
        except SandcastleTimeoutReached as ex:
            pass
        except ApiException as ex:
            pass
```

It's maybe a bit more verbose but more readable. (At least for me.)

**Comment by TomasTomecek** [comment]
ogr or packit are breaking the tests -_-

with last week's image the tests were passing locally

**Comment by TomasTomecek** [comment]
@lachmanfrantisek the last commit wouldn't happen if we gated packit PRs on p-s' test results - hence I'd say the commit message is pretty irrelevant since it's really about just fixing tests - which it clearly says, whatever the error was

**Comment by lachmanfrantisek** [comment]
> @lachmanfrantisek the last commit wouldn't happen if we gated packit PRs on p-s' test results - hence I'd say the commit message is pretty irrelevant since it's really about just fixing tests - which it clearly says, whatever the error was

I just wanted to point out that the commit (not the content) is unnecessary -- e.g. squash?

I am not against the gating of packit with p-s tests...;) I can find only [OGR issue](https://github.com/packit-service/ogr/issues/237), do we have one in packit?

**Comment by TomasTomecek** [comment]
> I just wanted to point out that the commit (not the content) is unnecessary -- e.g. squash?

but I'm 100% sure that those test fixes are not related to the main purpose of this PR, hence new commit

> I am not against the gating of packit with p-s tests...;) I can find only [OGR issue](https://github.com/packit-service/ogr/issues/237), do we have one in packit?

afaik we have it in jira in the testing epic and there is no upstream issue for that sadly :/

**Comment by lachmanfrantisek** [comment]
> but I'm 100% sure that those test fixes are not related to the main purpose of this PR, hence new commit
> 
> >

Sorry for nitpicking. I only don't like these dummy commits -- either add a meaningful message or squash. (I ask about that since we discuss that topic two weeks ago with the team..;)


> afaik we have it in jira in the testing epic and there is no upstream issue for that sadly :/

Here it is: https://github.com/packit-service/packit/issues/742

**Comment by TomasTomecek** [comment]
@lachmanfrantisek yes, I understand, but I still like to organize code so that commits represent "atomic" changes - fixing tests which got broken by something NOT related to my change is the exact content I don't want to have in the "main" commit; I mean even if I created a new PR with the test fixes, I'd have name it the same honestly :D but it's true that the commit message could have hold  the proper error message, I'll try to improve in future

so, more heated discussion on retro then :)

**Pull request merged**